Arrowgene.Ddon.GameServer/Handler/JobJobValueShopBuyItemHandler.cs

```diff
@@ -42,26 +42,19 @@ public override void Handle(GameClient client, StructurePacket<C2SJobJobValueSho
                     sendToItemBag = false;
                     break;
                 default:
-                    throw new Exception("Unexpected destination when buying goods: " + packet.Structure.StorageType);
+                    throw new ResponseErrorException(ErrorCode.ERROR_CODE_ITEM_INVALID_STORAGE_TYPE);
```

Comment:
You can still keep the string message like before for the server logs as a second argument in the exception constructor. The exception is so the client gets a error response packet with a value that'll get shown in their screen